src/mudlet-lua/lua/Other.lua

```diff
@@ -832,3 +806,66 @@ do
     end
   end
 end
+
+local timeframetable = {}
+
+function timeframe(vname, true_time, nil_time, ...)
+  assert(type(true_time) == "number" or type(true_time) == "table", "Timeframe requires a number or table argument as first parameter")
+  assert(type(nil_time) == "nil" or type(nil_time) == "number" or type(nil_time) == "table", "Timeframe requires a number or table if providing a second parameter")
+
+  local format = string.format
+	local vtype = type(vname)
+	-- aggregate timerlist data
+  local timerlist = {
+    {0, nil},
+    type(true_time) == "number" and {true_time, true} or type(true_time) == "table" and true_time,
+    type(nil_time) == "number" and {nil_time, nil} or type(nil_time) == "table" and nil_time,
+    ...
+  }
+
+	-- reinitialise timeframe for vname
+	killtimeframe(vname)
+  timeframetable[vname] = {}
+
+	-- loop through timerlist and create tempTimers
+  local maxtime = 0
+  for step, data in ipairs(timerlist) do
+    assert(type(data) == "table", "Wrong data type... Table required, got " .. type(data))
+    local time, value = data[1], data[2]
+
+    maxtime = (time > maxtime) and time or maxtime
+
+    local fun
+    if vtype == "function" then
+      fun = function()
+        local s,m = pcall(vname, value)
+        if not s then display(m) end
+      end
+    else
+      fun = assert(loadstring(format("%s = %s", vname, type(value) == "string" and ("'" .. value .. "'") or tostring(value))))
```

Comment:
Can you add an `assert` before this line (but within the `else` branch) to check for a `table`, `userdata` or `function`? Because those 3 types will most likely assign `nil` to the variable. But it should work correctly with the function version.

Reply:
Can you explain what you want me to do here, exactly? `fun` as far as I'm aware will always be a function call, so what variable are you checking types against?

Reply:
Well what would happen if `value` was a table, userdata or a function?

Reply:
The entire function would explode! No, I see your point now. So you want to assert to check that the value is *not* set to any of those types?

Reply:
Just assert that it is a type that is expected, with the nice error message [keneanung mentioned](https://github.com/Mudlet/Mudlet/pull/1360#discussion_r148649213). See more [examples here](https://github.com/Mudlet/Mudlet/blob/development/src/TLuaInterpreter.cpp#L5567).